pyrit/orchestrator/multi_turn/tree_of_attacks_with_pruning_orchestrator.py

```diff
@@ -0,0 +1,240 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+import random
+
+from pathlib import Path
+from treelib import Tree
+from typing import Optional
+
+from pyrit.common.path import DATASETS_PATH
+from pyrit.memory import MemoryInterface
+from pyrit.models import SeedPrompt
+from pyrit.orchestrator import MultiTurnOrchestrator, MultiTurnAttackResult
+from pyrit.prompt_converter import PromptConverter
+from pyrit.prompt_target import PromptTarget, PromptChatTarget
+from pyrit.score import SelfAskTrueFalseScorer, SelfAskScaleScorer, TrueFalseQuestion
+from pyrit.score.scorer import Scorer
+
+from pyrit.orchestrator.multi_turn.tree_of_attack_node import TreeOfAttackNode
+
+logger = logging.getLogger(__name__)
+
+
+class TapAttackResult(MultiTurnAttackResult):
+    def __init__(
+        self,
+        conversation_id: str,
+        achieved_objective: bool,
+        objective: str,
+        tree_visualization: Tree,
+    ):
+        super().__init__(
+            conversation_id=conversation_id,
+            achieved_objective=achieved_objective,
+            objective=objective,
+        )
+        self.tree_visualization = tree_visualization
+
+    def print_tree(self):
+        print(self.tree_visualization)
+
+
+class TreeOfAttacksWithPruningOrchestrator(MultiTurnOrchestrator):
+    _memory: MemoryInterface
+
+    def __init__(
+        self,
+        *,
+        objective_target: PromptTarget,
+        adversarial_chat: PromptChatTarget,
+        scoring_target: PromptChatTarget,
+        adversarial_chat_seed_prompt: Optional[SeedPrompt] = None,
+        adversarial_chat_system_prompt_path: Optional[Path] = None,
+        width: int = 3,
+        depth: int = 5,
+        branching_factor: int = 2,
+        on_topic_checking_enabled: bool = True,
+        prompt_converters: Optional[list[PromptConverter]] = [],
+        objective_achieved_score_threshhold: float = 0.8,
```

Comment:
```suggestion
        objective_achieved_score_threshold: float = 0.8,
```
nit: might as well clean it up even though it was there in the original code